kong/db/schema/init.lua

```diff
@@ -1141,37 +1183,18 @@ validate_fields = function(self, input)
   local errors, _ = {}
 
   local subschema = get_subschema(self, input)
+  local subschema_fields = subschema and subschema.fields or EMPTY
 
   for k, v in pairs(input) do
     local err
-    local field = self.fields[tostring(k)]
-
-    if field and field.type == "json" then
-      local json_schema = field.json_schema
-      local inline_schema = json_schema.inline
-
-      if inline_schema then
-        _, errors[k] = json_validate(v, inline_schema)
-
-      else
-        local parent_key = json_schema.parent_subschema_key
-        local json_subschema_key = input[parent_key]
-
-        if json_subschema_key then
-          local schema_name = json_schema.namespace .. "/" .. json_subschema_key
-          inline_schema = json.get_schema(schema_name) or json_schema.default
-
-          if inline_schema then
-            _, errors[k] = json_validate(v, inline_schema)
-
-          elseif not json_schema.optional then
-            errors[k] = validation_errors.JSON_SCHEMA_NOT_FOUND:format(schema_name)
-          end
+    k = tostring(k)
+    local field = self.fields[k]
+    local subschema_field = subschema_fields[k]
 
-        elseif not json_schema.optional then
-          errors[k] = validation_errors.JSON_PARENT_KEY_MISSING:format(k, parent_key)
-        end
-      end
+    if field and field.type == "json"
+      or (subschema_field and subschema_field.type == "json")
+    then
+      errors[k] = validate_json_field(subschema_field or field, k, input)
```

Comment:
The `plugins` entity makes use of subschemas to extend its schema/entity validation. Prior to this PR, it wasn't possible to validate `json`-typed fields that were members of a subschema, so that is the reason for this change.